operator/pkg/values/translation/translation_test.go

```diff
@@ -180,3 +181,382 @@ func TestGetTLSConfigValues(t *testing.T) {
 		})
 	}
 }
+
+func TestSetScannerComponentDisableValue(t *testing.T) {
+	tests := map[string]struct {
+		scannerComponent platform.ScannerComponentPolicy
+		want             chartutil.Values
+		wantErr          bool
+	}{
+		"Disabled": {
+			scannerComponent: platform.ScannerComponentDisabled,
+			want: chartutil.Values{
+				"disable": true,
+			},
+		},
+		"Enabled": {
+			scannerComponent: platform.ScannerComponentEnabled,
+			want: chartutil.Values{
+				"disable": false,
+			},
+		},
+		"Invalid": {
+			scannerComponent: "invalid",
+			want:             chartutil.Values{},
+			wantErr:          true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			vb := NewValuesBuilder()
+			scannerComponent := tt.scannerComponent
+			SetScannerComponentDisableValue(&vb, &scannerComponent)
+			values, err := vb.Build()
+			if tt.wantErr {
+				require.NotNil(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, values)
+		})
+	}
+}
+
+func TestSetScannerV4ComponentValues(t *testing.T) {
+	// using local copies to be able to create references to those constants
+	autoscalingEnabled := platform.ScannerAutoScalingEnabled
+	autoscalingDisabled := platform.ScannerAutoScalingDisabled
+	autoscalingInvalid := platform.AutoScalingPolicy("invalid")
+
+	tests := map[string]struct {
+		component    *platform.ScannerV4Component
+		componentKey string
+		want         chartutil.Values
+		wantErr      bool
+	}{
+		"empty for default component": {
+			component:    &platform.ScannerV4Component{},
+			componentKey: "indexer",
+			want:         chartutil.Values{},
+		},
+		"sets resources": {
+			component: &platform.ScannerV4Component{
+				DeploymentSpec: platform.DeploymentSpec{
+					Resources: &corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("200M"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("150m"),
+							corev1.ResourceMemory: resource.MustParse("250M"),
+						},
+					},
+				},
+			},
+			componentKey: "indexer",
+			want: chartutil.Values{
+				"indexer": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"cpu":    "100m",
+							"memory": "200M",
+						},
+						"limits": map[string]interface{}{
+							"cpu":    "150m",
+							"memory": "250M",
+						},
+					},
+				},
+			},
+		},
+		"uses componentKey as toplevel key": {
```

Comment:
I don't understand this label. Isn't this essentially the same test as the previous one, just for a different component (indexer - matcher) and with only a subset of the resource configuration being set?

Reply:
It tests that the function is not altered in a way that the top level key is not what's given as an argument.

I added it because all other tests only work with "indexer" as componentKey. I've adjusted the label, hope it's more clear.